src/System.Net.HttpListener/src/System/Net/HttpListener.Unix.cs

```diff
@@ -69,151 +80,305 @@ public AuthenticationSchemeSelector AuthenticationSchemeSelectorDelegate
+                    throw;
+                }
+                finally
+                {
+                    _state = State.Closed;
+                    if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
+                }
             }
         }
 
-        public ExtendedProtectionPolicy ExtendedProtectionPolicy
+        private void Close(bool force)
         {
-            get
+            CheckDisposed();
+            HttpEndPointManager.RemoveListener(this);
+            Cleanup(force);
+        }
+
+        internal void UnregisterContext(HttpListenerContext context)
+        {
+            lock ((_listenerContexts as ICollection).SyncRoot)
             {
-                throw new PlatformNotSupportedException();
+                _listenerContexts.Remove(context);
             }
-            set
+            lock ((_contextQueue as ICollection).SyncRoot)
             {
-                throw new PlatformNotSupportedException();
+                int idx = _contextQueue.IndexOf(context);
+                if (idx >= 0)
+                    _contextQueue.RemoveAt(idx);
             }
         }
 
-        public ServiceNameCollection DefaultServiceNames
+        internal void AddConnection(HttpConnection cnc)
         {
-            get
+            lock ((_connections as ICollection).SyncRoot)
             {
-                throw new PlatformNotSupportedException();
+                _connections[cnc] = cnc;
             }
         }
 
-        public string Realm
+        internal void RemoveConnection(HttpConnection cnc)
         {
-            get
+            lock ((_connections as ICollection).SyncRoot)
             {
-                throw new PlatformNotSupportedException();
+                _connections.Remove(cnc);
             }
-            set
+        }
+
+        internal void RegisterContext(HttpListenerContext context)
+        {
+            lock ((_listenerContexts as ICollection).SyncRoot)
             {
-                throw new PlatformNotSupportedException();
+                _listenerContexts[context] = context;
+            }
+
+            ListenerAsyncResult ares = null;
+            lock ((_asyncWaitQueue as ICollection).SyncRoot)
+            {
+                if (_asyncWaitQueue.Count == 0)
+                {
+                    lock ((_contextQueue as ICollection).SyncRoot)
+                        _contextQueue.Add(context);
+                }
+                else
+                {
+                    ares = _asyncWaitQueue[0];
+                    _asyncWaitQueue.RemoveAt(0);
+                }
+            }
+
+            if (ares != null)
+            {
+                ares.Complete(context);
             }
         }
 
-        public HttpListenerTimeoutManager TimeoutManager
+        private void Cleanup(bool close_existing)
         {
-            get
+            lock ((_listenerContexts as ICollection).SyncRoot)
```

Comment:
No specific reason, syncroot is a locking object for collections and these are internal objects.